modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerScheduleIndexRebuildTest.java

```diff
@@ -0,0 +1,595 @@
+
+        if (withCacheNames)
+            checkIndexes(CACHE_NAME_2_1);
+
+        ignite.close();
+
+        startGrid(0);
+
+        checkIndexes(CACHE_NAME_1_1);
+        checkIndexes(CACHE_NAME_1_2);
+
+        if (withCacheNames)
+            checkIndexes(CACHE_NAME_2_1);
+    }
+
+    /**
+     * Breaks sql index and checks that it is broken.
+     *
+     * @param ignite Node.
+     * @param cacheName Cache name.
+     * @throws Exception If failed.
+     */
+    private void breakAndCheckBroken(IgniteEx ignite, String cacheName) throws Exception {
+        injectTestSystemOut();
+
+        breakSqlIndex(ignite.cachex(cacheName), 1, null);
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "validate_indexes", "--check-crc", cacheName));
+
+        assertContains(log, testOut.toString(), "issues found (listed above)");
+
+        testOut.reset();
+    }
+
+    /**
+     * Checks that indexes are valid.
+     *
+     * @param cacheName Cache name.
+     */
+    private void checkIndexes(String cacheName) {
+        injectTestSystemOut();
+
+        assertEquals(EXIT_CODE_OK, execute("--cache", "validate_indexes", "--check-crc", cacheName));
+
+        assertContains(log, testOut.toString(), "no issues found.");
+
+        testOut.reset();
+    }
+
+    /**
+     * Waits for the rebuild of the indexes.
+     *
+     * @param ignite Ignite instance.
+     * @return {@code True} if index rebuild was completed before {@code timeout} was reached.
+     * @throws IgniteInterruptedCheckedException if failed.
+     */
+    private boolean waitForIndexesRebuild(IgniteEx ignite) throws IgniteInterruptedCheckedException {
+        return GridTestUtils.waitForCondition(
+            () -> ignite.context().cache().publicCaches()
+                .stream()
+                .allMatch(c -> c.indexReadyFuture().isDone()),
+            REBUILD_TIMEOUT);
+    }
+
+    /**
+     * Checks that given indexes are scheduled for the rebuild.
+     *
+     * @param node Node.
+     * @param cacheToIndexes Map of caches to indexes.
+     */
+    private void checkIndexesRebuildScheduled(IgniteEx node, Map<Integer, Set<String>> cacheToIndexes) {
+        MaintenanceTask maintenanceTask = node.context().maintenanceRegistry().requestedTask(INDEX_REBUILD_MNTC_TASK);
+
+        assertNotNull(maintenanceTask);
+
+        List<MaintenanceRebuildIndexTarget> targets = parseMaintenanceTaskParameters(maintenanceTask.parameters());
+
+        Map<Integer, Set<String>> result = targets.stream().collect(groupingBy(
+            MaintenanceRebuildIndexTarget::cacheId,
+            mapping(MaintenanceRebuildIndexTarget::idxName, toSet())
+        ));
+
+        assertEqualsMaps(cacheToIndexes, result);
+    }
+
+    /**
+     * Returns indexes' names of the given cache.
+     *
+     * @param node Node.
+     * @param cache Cache name.
+     * @return Indexes of the cache.
+     */
+    private Set<String> indexes(IgniteEx node, String cache) {
```

Comment:
I think this code is duplicated. Could we extract it to some util class to avoid duplication?